volatility/framework/automagic/pdbscan.py

```diff
@@ -310,10 +146,12 @@ def method_fixed_mapping(self,
         physical_layer_name = self.get_physical_layer_name(context, vlayer)
         kvo_path = interfaces.configuration.path_join(vlayer.config_path, 'kernel_virtual_offset')
 
-        kernels = scan(ctx = context,
-                       layer_name = physical_layer_name,
-                       page_size = vlayer.page_size,
-                       progress_callback = progress_callback)
+        kernel_pdb_names = [bytes(name + ".pdb", "utf-8") for name in constants.windows.KERNEL_MODULE_NAMES]
```

Comment:
The three discovery methods seems they could use a little cleanup to be more understandable and less repeated code:

1) method_module_offset & method_kdbg_offset seem to be copy/paste of each other except for the string searched and how much is subtracted from the address of the string found to the containing data structure. An internal function that took those values as parameters would collapse the the current methods to one line each

2) method_fixed_mapping:
 - The check of "or not isinstance(kernel['mz_offset'], int)" should be removed as mz_offset can only be None or the result of multiplying two integers.

- this method sets kernel_virtual_offset itself whereas the other two let set_kernel_virtual_offset set it. From reading the code, it seems like the same value will be eventually set, so the duplication be should be removed.

Reply:
1. Cool, I'll give it a look
2. 
* The check may be there to help mypy (which otherwise complains the value may not be an int because it's not clever enough to figure that out), but I'll check the git blame to see why it's there and add a comment if it's mypy related, or remove it if not.
* Sure, I can look into that code and improve it.

Reply:
Yeah, 2a was a typing thing, so I'm going to leave it in.  The static type checker can only figure out types like that with something like an explicit `isinstance`.  Eventually, when it gets smarter, we can take the redundant check out.